os/net/app-layer/mqtt/mqtt.h

```diff
@@ -544,14 +746,45 @@ void mqtt_set_username_password(struct mqtt_connection *conn,
 void mqtt_set_last_will(struct mqtt_connection *conn,
                         char *topic,
                         char *message,
+#if MQTT_5
+                        mqtt_qos_level_t qos,
+                        struct mqtt_prop_list_t *will_props);
+#else
                         mqtt_qos_level_t qos);
+#endif
 
 #define mqtt_connected(conn) \
   ((conn)->state == MQTT_CONN_STATE_CONNECTED_TO_BROKER ? 1 : 0)
 
 #define mqtt_ready(conn) \
   (!(conn)->out_queue_full && mqtt_connected((conn)))
 /*---------------------------------------------------------------------------*/
+void encode_var_byte_int(uint8_t *vbi_out,
+                         uint8_t *vbi_bytes,
+                         uint32_t val);
+/*---------------------------------------------------------------------------*/
+uint8_t decode_var_byte_int(const uint8_t *input_data_ptr,
+                            int input_data_len,
+                            uint32_t *input_pos,
+                            uint32_t *pkt_byte_count,
+                            uint16_t *dest);
+/*---------------------------------------------------------------------------*/
+#if MQTT_5
+/**
+ * \brief Send authentication message (MQTTv5-only).
+ * \param conn A pointer to the MQTT connection.
+ * \param auth_type The type of auth to send (continue authentication or
+ *        re-authentication).
+ * \param prop_list Output properties.
+ * \return MQTT_STATUS_OK or some error status
+ *
+ * This function send an MQTT authentication message.
+ */
+mqtt_status_t mqtt_auth(struct mqtt_connection *conn,
+                        mqtt_auth_type_t auth_type,
+                        struct mqtt_prop_list_t *prop_list);
+#endif
```

Comment:
OK functionality only supported in v5, but perhaps the prototype does not have to be conditional?